syntax = "proto3";

package flowpipe.stages.csv.sink.v1;

// Configuration for CsvSink stage
message CsvSinkConfig {
  // File path to write
  string path = 1;

  // CSV delimiter (default: ",")
  string delimiter = 2;

  // Include header row on first write
  bool include_header = 3;

  // Append to existing file instead of truncating
  bool append = 4;

  // Optional header ordering. If empty, headers are derived from payload.
  repeated string headers = 5;

  // Quote all fields instead of only those requiring escaping
  bool quote_all_fields = 6;
}
